File: source/aws/py/lambda-websocket/lambda_entry.py
# use the '../.env' environment variables for local debugging
import use_env
import asyncio
from typing import Dict
from UniRpc.LambdaWebsocketTypes import IWebsocketEvent
from UniRpc.WebsocketService import WebsocketService


async def websocket_handler(event: IWebsocketEvent):
    websocketService: WebsocketService = WebsocketService()
    await websocketService.ProcessEvent(event)


def handler(event: IWebsocketEvent, context: Dict[str, any] = None):
    asyncio.run(websocket_handler(event))
    return {'status': 200}


if __name__ == '__main__':
    handler({})

File: source/aws/ts/lambda-websocket/lambda_entry.ts
import './use_env';
import {} from 'aws-sdk';
import { IWebsocketEvent } from './UniRpc/LambdaWebsocketTypes';
import { WebsocketService } from './UniRpc/WebsocketService';

export async function handler (event: IWebsocketEvent) {
    console.log('AWS Lambda Project "@{project}".');
    let websocketService = new WebsocketService();
    // Register your service implementations here.
    // websocketService.RegisterService(new Namespace.ServiceImpl());
    let response = await websocketService.ProcessEvent(event);
    console.log('Function Response:', response);
    return response;
};

// Entry point fo the process
if (require.main == module) {
    (async () => await handler({} as any))();
}